File: src/app/modules/m1-school-office/data-access/models/school-accreditation-log.ts
import type { RecordStatus, LicenseType } from './common';

export interface SchoolAccreditationLog {
  id: number;
  schoolId: number;
  licenseNumber: string;
  accreditationBody: string;
  issueDate: string;
  expiryDate: string;
  status: RecordStatus;
  licenseType: LicenseType;
  accreditationGrade: string | null;
  certificateAttachmentUrl: string | null;
  renewalSubmittedDate: string | null;
}

export type CreateSchoolAccreditationLogDto = Omit<SchoolAccreditationLog, 'id' | 'status'>;

export type UpdateSchoolAccreditationLogDto = Omit<SchoolAccreditationLog, 'schoolId' | 'status'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolAccreditationLog.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class SchoolAccreditationLog : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string LicenseNumber { get; set; } = string.Empty;
    public string AccreditationBody { get; set; } = string.Empty;
    public DateTime IssueDate { get; set; }
    public DateTime ExpiryDate { get; set; }
    public int Status { get; set; } // 1=Active, 2=Expired, 3=Suspended
    public int LicenseType { get; set; } = 1; // 1=Academic, 2=Safety, 3=Municipal, 4=Ministry
    public string? AccreditationGrade { get; set; }
    public string? CertificateAttachmentUrl { get; set; }
    public DateTime? RenewalSubmittedDate { get; set; }

    // Navigation Property
    public virtual School? School { get; set; }
}
